src/backend/distributed/planner/multi_join_order.c

```diff
@@ -1359,11 +1358,8 @@ DistPartitionKey(Oid relationId)
 		return NULL;
 	}
 
-	/* now obtain partition key and build the var node */
-	variableNode = stringToNode(partitionEntry->partitionKeyString);
-
-	partitionKey = (Var *) variableNode;
-	Assert(IsA(variableNode, Var));
+	partitionKey = copyObject(partitionEntry->partitionColumn);
```

Comment:
When `__typeof__` is supported, this will actually expand to `(Var *) copyObjectImpl(partitionEntry->partitionColumn)`. I think the `Assert` is kind of redundant here.

Reply:
(I suppose this doesn't actually ensure the thing it points to is a `Var` in the PostgreSQL sense, but looking at the impl it's still pretty redundant. I think we should `Assert` on what `stringToNode` returns and then assume this is a `Var` from then on out).